### PR TITLE
[master] [SERV-952] Added section for Codehaus Cargo plugin

### PR DESCRIPTION
**Pull request by felixif**
Added a new section describing the usage of Codehaus Cargo with Payara Server as described in SERV-952.

### CODE REVIEW

File: enterprise/docs/modules/ROOT/pages/Technical Documentation/Payara Server Documentation/Extensions/Overview.adoc [hunk 1]
```diff
@@ -5,4 +5,7 @@ Payara Server Enterprise Edition allows plugin extensions to expand the features
 Currently, the following extensions are provided:
 
 xref:/Technical Documentation/Payara Server Documentation/Extensions/gRPC Support/Overview.adoc[gRPC Support]::
-The gRPC Support module allows web applications to define gRPC services which can be consumed by client applications.
\ No newline at end of file
+The gRPC Support module allows web applications to define gRPC services which can be consumed by client applications.
+
+xref:/Technical Documentation/Payara Server Documentation/Extensions/Codehaus Cargo/Overview.adoc[Codehaus Cargp]::
```
Comment (JamesHillyard) [inline enterprise/docs/modules/ROOT/pages/Technical Documentation/Payara Server Documentation/Extensions/Overview.adoc:10]:
```suggestion
xref:/Technical Documentation/Payara Server Documentation/Extensions/Codehaus Cargo/Overview.adoc[Codehaus Cargo]::
```

File: enterprise/docs/modules/ROOT/pages/Technical Documentation/Payara Server Documentation/Extensions/Codehaus Cargo/Overview.adoc [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+        <dependency>
+            <groupId>org.glassfish.main.deployment</groupId>
+            <artifactId>deployment-client</artifactId>
+            <version>5.0</version>
+        </dependency>
+    </dependencies>
+</plugin>
+----
+
+=== Remote deployment
```
Comment (JamesHillyard) [inline enterprise/docs/modules/ROOT/pages/Technical Documentation/Payara Server Documentation/Extensions/Codehaus Cargo/Overview.adoc:134]:
```suggestion
[[remote-deployment]]
=== Remote Deployment
```

File: enterprise/docs/modules/ROOT/pages/Technical Documentation/Payara Server Documentation/Extensions/Codehaus Cargo/Overview.adoc [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+|cargo:deployer-deploy (aliased to cargo:deploy)
+|Deploy a deployable to a running container.
+
+The cargo:start and cargo:run do already deploy the deployables specified in the configuration to the container; as a result calling cargo:deploy for a container which has been started by Codehaus Cargo in the same Maven project will most likely cause a second deployment of the same deployables (and might even fail).
+
+|cargo:deployer-undeploy (aliased to cargo:undeploy)
+|Undeploy a deployable from a running container.
+|===
+
+=== Local deployment
```
Comment (JamesHillyard) [inline enterprise/docs/modules/ROOT/pages/Technical Documentation/Payara Server Documentation/Extensions/Codehaus Cargo/Overview.adoc:67]:
```suggestion
[[local-deployment]]
=== Local Deployment
```

File: enterprise/docs/modules/ROOT/pages/Technical Documentation/Payara Server Documentation/Extensions/Codehaus Cargo/Overview.adoc [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+            <artifactId>cargo-maven3-plugin</artifactId>
+            <version>1.10.6</version>
+        </plugin>
+        ...
+    </plugins>
+</build>
+----
+
+[[execution-goals]]
+=== Execution goals
```
Comment (JamesHillyard) [inline enterprise/docs/modules/ROOT/pages/Technical Documentation/Payara Server Documentation/Extensions/Codehaus Cargo/Overview.adoc:39]:
```suggestion
=== Execution Goals
```

File: enterprise/docs/modules/ROOT/pages/Technical Documentation/Payara Server Documentation/Extensions/Codehaus Cargo/Overview.adoc [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+Cargo is a thin wrapper that allows you to manipulate various types of application containers (J2EE, Java EE, Jakarta EE and others) in a standard way. The portion Codehaus is a reference to the open-source community which used to host Cargo when it was first created. Since then, the community has been moved to GitHub.
+
+[[what-can-we-do-with-it]]
+== What can we do with it?
+
+* Start, stop or restart an application container
+* Deploy, undeploy or redeploy an application on local or remote instances
+
+[[how-to-use-it]]
+== How to use it?
```
Comment (JamesHillyard) [inline enterprise/docs/modules/ROOT/pages/Technical Documentation/Payara Server Documentation/Extensions/Codehaus Cargo/Overview.adoc:16]:
```suggestion
[[how-to-use-codehaus-cargo]]
== How to Use Codehaus Cargo
```

File: enterprise/docs/modules/ROOT/pages/Technical Documentation/Payara Server Documentation/Extensions/Codehaus Cargo/Overview.adoc [hunk 6]
```diff
@@ -0,0 +1,189 @@
+[[codehaus-cargo]]
+= Codehaus Cargo
+
+[[what-is-codehaus-cargo]]
+== What is Codehaus Cargo?
+
+Cargo is a thin wrapper that allows you to manipulate various types of application containers (J2EE, Java EE, Jakarta EE and others) in a standard way. The portion Codehaus is a reference to the open-source community which used to host Cargo when it was first created. Since then, the community has been moved to GitHub.
+
+[[what-can-we-do-with-it]]
+== What can we do with it?
```
Comment (JamesHillyard) [inline enterprise/docs/modules/ROOT/pages/Technical Documentation/Payara Server Documentation/Extensions/Codehaus Cargo/Overview.adoc:10]:
```suggestion
Codehaus Cargo allows you to::
```

File: enterprise/docs/modules/ROOT/pages/Technical Documentation/Payara Server Documentation/Extensions/Codehaus Cargo/Overview.adoc [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+            <version>1.10.6</version>
+        </plugin>
+        ...
+    </plugins>
+</build>
+----
+
+[[execution-goals]]
+=== Execution goals
+In the following table, you can find the list of the main goals provided by the Cargo plugin. For more information about the goals, please refer to the link:https://codehaus-cargo.github.io/cargo/Maven3+plugin.html[Cargo Documentation - Maven 3 Plugin].
```
Comment (JamesHillyard) [inline enterprise/docs/modules/ROOT/pages/Technical Documentation/Payara Server Documentation/Extensions/Codehaus Cargo/Overview.adoc:40]: This link is dead

**Review by JamesHillyard — CHANGES_REQUESTED** [review]
Is this actually enterprise only? I don't see why this wouldn't also work in Community

**Review by JamesHillyard — APPROVED** [review]
This should still probably be available in the community docs (Unless I'm missing something?) - But the content looks good

**Comment by fturizo** [comment]
@JamesHillyard  is correct, the documentation applies to both editions. I also moved the documentation to the Ecosystem section since it makes more sense to be there since the plugin is not an official extension of Payara Server.

**Comment by fturizo** [comment]
@felixif, please make an equivalent PR for the Payara 6 branch.

**Pull request merged**